numpy/ma/core.py

```diff
@@ -7683,6 +7676,10 @@ def asanyarray(a, dtype=None):
     <class 'numpy.ma.core.MaskedArray'>
 
     """
+    # workaround for #8666, to preserve identity. Ideally the bottom line
```

Comment:
One could change `MaskedArray.__new__`, though perhaps calling a class constructor and not getting back a new instance might also be surprising... Perhaps best might have been to have had a logical distinction between the `np.ma.MaskedArray` class and a `np.ma.masked_array`/`np.ma.array` function...

Overall, this is probably the best solution.